exporter/stats/prometheus/prometheus.go

```diff
@@ -170,14 +170,14 @@ func (c *collector) Describe(ch chan<- *prometheus.Desc) {
 // Collect is invoked everytime a prometheus.Gatherer is run
 // for example when the HTTP endpoint is invoked by Prometheus.
 func (c *collector) Collect(ch chan<- prometheus.Metric) {
-	c.mu.Lock()
-	views := make(map[string]*stats.ViewData, len(c.viewData))
-	for i, vd := range c.viewData {
-		views[i] = vd
-	}
-	c.mu.Unlock()
-
 	for _, vd := range c.viewData {
```

Comment:
Can we reject them at Export instead?

Reply:
I was following suit with https://github.com/census-instrumentation/opencensus-go/blob/9b9df23b5a299ba044805cdd2c28764702cfddab/exporter/stats/stackdriver/stackdriver.go#L168-L172 and the suggestion from the issue originally, but for sure, it even makes it easier to reject them in the Export method.